autoscheduler/scraper/models/department.py

```diff
@@ -1,9 +1,17 @@
 from django.db import models
 
+def generate_department_id(code: str, term: str):
+    """ Generates the department id as {code}-{term} """
+
+    return "".join((code, '-', term))
+
 class Department(models.Model):
     """ Data structure that stores the department tag and description """
-    code = models.CharField(max_length=4, primary_key=True) # Ex: CSCE
+
+    id = models.CharField(max_length=10, primary_key=True) # code + term
+    code = models.CharField(max_length=4) # Ex: CSCE
     description = models.TextField(max_length=100) # Ex: CSCE - Computer Sci & Engr
+    term = models.CharField(max_length=6, db_index=True)

```

Comment:
There are different departments for a given semester(for instance, there are 45 more departments in fall 2019 then fall 2013)